NickvisionCavalier.GNOME/Views/PreferencesDialog.cs

```diff
@@ -742,11 +779,105 @@ private async Task AddColor(ColorType type)
         catch { }
     }
 
+    /// <summary>
+    /// Occurs when color button was clicked
+    /// </summary>
+    /// <param name="sender">Color box that should be edited</param>
+    /// <param name="e">Color args for controller</param>
     private void OnEditColor(object sender, ColorEventArgs e) => _controller.EditColor(e.Type, e.Index, e.Color);
 
+    /// <summary>
+    /// Occurs when delete color button was clicked
+    /// </summary>
+    /// <param name="sender">Color box that should be deleted</param>
+    /// <param name="e">Color args for controller</param>
     private void OnDeleteColor(object sender, ColorEventArgs e)
     {
         _controller.DeleteColor(e.Type, e.Index);
         UpdateColorsGrid();
     }
+
+    /// <summary>
+    /// Update flowbox with images
+    /// </summary>
+    public void UpdateImagesList()
+    {
+        var paths = _controller.GetImagesList();
+        if (paths.Count == 0)
+        {
+            _imagesStack.SetVisibleChildName("empty");
+            return;
+        }
+        _imagesStack.SetVisibleChildName("images");
+        _removingImages = true;
+        while (_imagesFlowBox.GetFirstChild() != null)
+        {
+            _imagesFlowBox.Remove(_imagesFlowBox.GetFirstChild()!);
+        }
+        _removingImages = false;
+        var emptyImage = Gtk.Image.NewFromIconName("x-circular-symbolic");
+        emptyImage.SetTooltipText(_("No Image"));
+        emptyImage.AddCssClass("cavalier-image");
+        emptyImage.SetSizeRequest(160, 120);
+        emptyImage.SetPixelSize(42);
+        emptyImage.SetMarginTop(2);
+        emptyImage.SetMarginStart(2);
+        emptyImage.SetMarginEnd(2);
+        emptyImage.SetMarginBottom(2);
+        _imagesFlowBox.Append(emptyImage);
+        for (var i = 0; i < paths.Count; i++)
+        {
+            var image = new ImageItem(paths[i], i);
+            image.OnRemoveImage += RemoveImage;
+            _imagesFlowBox.Append(image);
+        }
+        try
+        {
+            _imagesFlowBox.SelectChild(_imagesFlowBox.GetChildAtIndex(_controller.ImageIndex)!);
+        }
+        catch (IndexOutOfRangeException)
+        {
+            _imagesFlowBox.SelectChild(_imagesFlowBox.GetChildAtIndex(0)!);
+        }
+        _imageScale.SetValue(_controller.ImageScale);
+    }
+
+    /// <summary>
+    /// Add an image to the images list
+    /// </summary>
+    public async Task AddImageAsync()
+    {
+        try
```

Comment:
I would only try the `OpenAsync` part as thats the only thing known to throw an exception...